### ISSUE TITLE
Cancel running jobs when new ones are started for a PR

### ISSUE THREAD

**Issue by TomasTomecek** [packit/packit-service#5]
It can easily happen that a PR can receive a push or a `/packit copr-build` command and then another one in a short time-span.

In order to not to waste resources and avoid publishing results from a parallel set of jobs into the same PR, we should cancel old jobs (builds, tests) and submit new ones.

**Comment by TomasTomecek** [comment]
I wonder if copr optimizes this on its own - cancels old builds if new are requested in a project.

**Comment by TomasTomecek** [comment]
with postgres this can be now done so easily honestly

**Comment by TomasTomecek** [comment]
we should still investigate if copr can do this on it own

**Comment by lachmanfrantisek** [comment]
Related to #785

**Comment by lachmanfrantisek** [comment]
Still nice to have.

**Comment by jpopelka** [comment]
Can we bump priority here? @mrc0mmand mentioned this again when talking about expectations ("Error Budget").

**Comment by lachmanfrantisek** [comment]
Yes, sure. I am adding the issue to the project board so we can groom and plan for the next sprint(s).

**Comment by csomh** [comment]
We discussed the following approach today:
- Upgrade the pipeline-abstraction so that it's possible to tell if a pipeline is for a PR or branch is still in progress.
- Add a new celery task at the beginning of the pipeline which would check if there is an another pipeline already running for the PR or branch, and cancel the that pipeline.
- Cancelling the pipeline means:
  - terminating SRPM builds if they are running
  - terminating Copr builds if they are running
  - terminating TF tests if they are running
  - internally marking the pipeline as cancelled
  - (GitHub flags should be already updated by this point, so we should just make sure they are not overwritten by the termination process).

**Comment by lachmanfrantisek** [comment]
Let's start with something more easier and doable in short-term:

* [ ] Terminate an old Copr build when submitting a new one.

**Comment by lbarcziova** [comment]
We discussed this and decided we will drop this issue since we are not able to plan this in the near future and the implementation could be more complex (since the codebase changed a lot since opening this issue) and lead to race conditions.

**Comment by lbarcziova** [comment]
Reopening given the increased user base and since it would help with resource management and address challenges such as handling outages where queues become overloaded. This would prevent unnecessary triggers of builds/tests when a pull request is updated during such outages.

This was also asked by @evgeni in the chat recently.

**Comment by bocekm** [comment]
+1 to this RFE!

When I trigger copr building and integration test execution on https://github.com/oamg/convert2rhel PRs, that's a lot of jobs. When I trigger that all again, for example by pushing to the PR branch, before even copr builds are created then the Testing Farm jobs that follow up after copr builds don't need to be even run. That might even save some GitHub App requests which are limited and our project has been hiting the limit https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?primary-rate-limit-for-github-app-installations=&apiVersion=2022-11-28#primary-rate-limit-for-github-app-installations.
